tests/test_indexing.py

```diff
@@ -1,12 +1,23 @@
+"""Test windows and indexing"""
+
+# TODO: break up multi-assertion tests.
+
 import numpy as np
 import pytest
 
 import rasterio
 from rasterio import windows
 
+
 DATA_WINDOW = ((3, 5), (2, 6))
 
 
+def assert_window_almost_equals(a, b, precision=6):
```

Comment:
why not use numpy's `allclose` instead?  You are already using numpy in `windows` so it is not an issue to bring it in here...